test/core/transport/chaotic_good/client_transport_error_test.cc

```diff
@@ -0,0 +1,427 @@
+                      GRPC_STATUS_UNAVAILABLE);
+            EXPECT_TRUE(std::get<1>(ret).ok());
+            return absl::OkStatus();
+          }),
+      EventEngineWakeupScheduler(event_engine_),
+      [&on_done](absl::Status status) { on_done.Call(std::move(status)); });
+  // Wait until ClientTransport's internal activities to finish.
+  event_engine_->TickUntilIdle();
+  event_engine_->UnsetGlobalHooks();
+}
+
+TEST_F(ClientTransportTest, AddMultipleStreamWithReadFailed) {
+  // Mock read failed at first stream, and second stream's write will fail too.
+  EXPECT_CALL(control_endpoint_, Read)
+      .InSequence(control_endpoint_sequence_)
+      .WillOnce(WithArgs<0>(
+          [](absl::AnyInvocable<void(absl::Status)> on_read) mutable {
+            on_read(absl::InternalError("control endpoint read failed."));
+            // Return false to mock EventEngine read not finish.
+            return false;
+          }));
+  InitialClientTransport();
+  ClientMetadataHandle first_stream_md;
+  ClientMetadataHandle second_stream_md;
+  auto first_stream_args =
+      CallArgs{std::move(first_stream_md),
+               ClientInitialMetadataOutstandingToken::Empty(),
+               nullptr,
+               &pipe_server_intial_metadata_.sender,
+               &pipe_client_to_server_messages_.receiver,
+               &pipe_server_to_client_messages_.sender};
+  auto second_stream_args =
+      CallArgs{std::move(second_stream_md),
+               ClientInitialMetadataOutstandingToken::Empty(),
+               nullptr,
+               &pipe_server_intial_metadata_second_.sender,
+               &pipe_client_to_server_messages_second_.receiver,
+               &pipe_server_to_client_messages_second_.sender};
+  StrictMock<MockFunction<void(absl::Status)>> on_done;
+  EXPECT_CALL(on_done, Call(absl::OkStatus()));
+  auto activity = MakeActivity(
+      Seq(
+          // Concurrently: write and read messages from client transport.
+          Join(
+              // Add first stream with call_args into client transport.
+              AddStream(std::move(first_stream_args)),
+              // Send messages to first stream's
+              // call_args.client_to_server_messages pipe, which will be
+              // eventually sent to control/data endpoints.
+              SendClientToServerMessages(pipe_client_to_server_messages_, 1)),
+          Join(
+              // Add second stream with call_args into client transport.
+              AddStream(std::move(second_stream_args)),
+              // Send messages to second stream's
+              // call_args.client_to_server_messages pipe, which will be
+              // eventually sent to control/data endpoints.
+              SendClientToServerMessages(pipe_client_to_server_messages_second_,
+                                         1)),
+          // Once complete, verify successful sending and the received value.
+          [](const std::tuple<ServerMetadataHandle, absl::Status>& ret) {
```

Comment:
Same comment as above w.r.t adding a new blob to assert the result of the first Join in line 393.

Reply:
Done, thanks for the suggestion.